installer/dev/install.cpp

```diff
@@ -224,22 +238,87 @@ namespace WindowsAppRuntimeInstaller {
             if (!quiet)
             {
-                std::wcout << "Provisioning result : 0x" << std::hex << hrProvisionResult << std::endl;
+                std::wcout << "Provisioning result : 0x" << std::hex << hrProvisionResult << " ";
+                ShowErrorMessage(hrProvisionResult);
             }
             LOG_IF_FAILED(hrProvisionResult);
         }
+    }
 
-        return;
+    HRESULT Deploy(const WindowsAppRuntimeInstaller::Options options) noexcept try
+    {
+        RETURN_IF_FAILED(DeployPackages(options));
+        RETURN_IF_FAILED(InstallLicenses(options));
+        return S_OK;
     }
+    CATCH_RETURN()
 
-    HRESULT DeployPackages(bool quiet) noexcept try
+    HRESULT DeployPackages(const WindowsAppRuntimeInstaller::Options options)
     {
-        for (const auto& package : WindowsAppRuntimeInstaller::c_packages)
+        if (WI_IsFlagSet(options, WindowsAppRuntimeInstaller::Options::InstallPackages))
         {
-            DeployPackageFromResource(package, quiet);
+            for (const auto& package : WindowsAppRuntimeInstaller::c_packages)
+            {
+                DeployPackageFromResource(package, options);
+            }
         }
+        return S_OK;
+    }
+
+    HRESULT InstallLicenses(const WindowsAppRuntimeInstaller::Options options)
+    {
+#if defined(WAR_PROCESS_LICENSES)
+        const auto quiet{ WI_IsFlagSet(options, WindowsAppRuntimeInstaller::Options::Quiet) };
 
+        if (WI_IsFlagSet(options, WindowsAppRuntimeInstaller::Options::InstallLicenses))
+        {
+            Microsoft::Windows::ApplicationModel::Licensing::Installer licenseInstaller;
+            for (const auto& license : WindowsAppRuntimeInstaller::c_licenses)
+            {
+                if (!quiet)
+                {
+                    std::wcout << "Installing license: " << license.id << std::endl;
+                }
+
+                // DryRun = Don't the work
+                if (WI_IsFlagSet(options, WindowsAppRuntimeInstaller::Options::DryRun))
+                {
+                    continue;
+                }
+
+                // Install the license
+                auto thisModule{ reinterpret_cast<HINSTANCE>(&__ImageBase) };
+                const auto hr{ licenseInstaller.InstallLicense(thisModule, license.id) };
+                if (!quiet)
+                {
+                    std::wcout << "Install result : 0x" << std::hex << hr << " ";
+                    ShowErrorMessage(hr);
+                }
+                RETURN_IF_FAILED_MSG(hr, "License:%ls", license.id.c_str());
+            }
+        }
+#endif
         return S_OK;
     }
-    CATCH_RETURN()
+
+    void ShowErrorMessage(const HRESULT hr)
+    {
+        if (SUCCEEDED(hr))
+        {
+            std::wcout << std::endl;
+            return;
+        }
+
+        PWSTR message{};
+        if (FormatMessage(FORMAT_MESSAGE_ALLOCATE_BUFFER | FORMAT_MESSAGE_FROM_SYSTEM,
```

Comment:
Seems like a good WIL helper to add, maybe file an issue to track?

Reply:
Was thinking same as I was cribbing it (actually surprised it's not already in WIL).

Ask tracked via https://task.ms/37017993